### PR TITLE
[prometheus-windows-exporter] Add chart

### PR DESCRIPTION
**Pull request by jkroepke**
#### What this PR does / why we need it

Add helm chart for windows export.

Tested on a AKS with WS 2022 nodes.

image

image

Based on https://github.com/prometheus-community/windows_exporter/blob/master/kubernetes/kubernetes.md

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-windows-exporter/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v2
+name: prometheus-windows-exporter
+description: A Helm chart for prometheus windows-exporter
+keywords:
+  - node-exporter
+  - prometheus
+  - exporter
+type: application
+version: 1.0.0
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply charts/prometheus-windows-exporter/Chart.yaml]: Done

File: charts/prometheus-windows-exporter/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+```console
+helm uninstall [RELEASE_NAME]
+```
+
+This removes all the Kubernetes components associated with the chart and deletes the release.
+
+_See [helm uninstall](https://helm.sh/docs/helm/helm_uninstall/) for command documentation._
+
+## Upgrading Chart
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply charts/prometheus-windows-exporter/README.md]: Done

**Review by monotek — COMMENTED** [review]
How do we actualy test this?
Is the exporter binary not a windows exe file and we would need windows nodes?

**Comment by jkroepke** [comment]
In theory, yet. We need a Linux node for the control plane (because there are not windows container for the Kubernetes control plane) and a windows nodes for the binary itself. It can be archived through WSL....

I guess the complexity of such an test environment are high that no one could maintain this here.

tbh, we can not test this in GH Actions.

**Comment by jkroepke** [comment]
@monotek But we are still testing, if the manifests are valid and does not contain any error. However the DaemonSet will have replica 0 on the GH action pipelines. Theoretically, it chart itself is tested.

**Comment by jkroepke** [comment]
I'm plan to use the helm chart in one of our managed service project. 

@lippertmarkus if you have the time, maybe you can also this chart on your infrastructure?

**Comment by jkroepke** [comment]
@monotek how to move forward here?

**Comment by monotek** [comment]
I don't have any windows nodes.

**Comment by monotek** [comment]
So you have testet the chart now on your windows nodes?

**Comment by jkroepke** [comment]
@monotek Yes. The screenshot form first post document that it works.

I setup an AKS cluster in azure with windows nodes on a personal subscription from my company. With AKS, it's a no brainer to setup windows nodes.

**Review by monotek — APPROVED** [review]

**Pull request merged**